### DISCUSSION TITLE
Plug-in single measurement

### DISCUSSION THREAD

**Discussion by EricLjr** [apache/skywalking#8916] [Q&A]
**Could you please share how Skywalking plugins are tested?**
Can you give the single test list steps?
For example, how does the mysqlClinet plug-in test to belong to pass? I have participated in the development of mysqlClient plug-in before. Now I would like to sort out the single test process and hope to get relevant advice. Thank you very much!

**Comment by wu-sheng** [comment]
Are you checking the upstream(you want to submit a pull request) test process? Or something you want to do locally and privately?

**Reply by EricLjr** [reply]
Thank you for your answer.
I want to locally imitate the single test of Skywalking plug-in and learn from the single test process framework. Because I think your single test process is very standard

**Reply by wu-sheng** [reply]
You could find the test tool with doc from here, https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/java-plugin-development-guide/#plugin-test-tool
